app/models/concerns/credit_card.rb

```diff
@@ -0,0 +1,57 @@
+module CreditCard
```

Comment:
I'm not sure if this module is necessary. Most of the behavior here is about validation concerns which should stay in the validator, the only thing the model really cares about is the franchise. It's a pretty small change, but I think something like this may do a better job keeping those concerns separate

```
class Contact < ApplicationRecord
  ...
  validates :credit_card_franchise, presence: true
  validates :credit_card, credit_card: true, presence: true, if: :credit_card_franchise?
  ...

  def credit_card_franchise
     @credit_card_franchise ||= #method implementation
  end
end

# CreditCardValidator would look pretty much the same, but using `record.credit_card_franchise` 
# instead of having a determine_franchise method.
```